### ISSUE TITLE
Manage selections while mutation of node

### ISSUE THREAD

**Issue by HeinPauwelyn** [code4me/4me-sdk-graphql-dotnet#4]
We want to do a mutation to add a note to a request and as response the notes must be send back. For this only the permissions request:Update, request:Read and note:Read are needed using Postman.

```graphQl
mutation {
    requestUpdate (
        input: {
            id: "NP2inFhLJlcS83NjQ0ODpM5dasU4Fake"
            note: "This is a note"
        }
    ) {
        request {
            id
            subject
            notes (first: 10) {
                nodes {
                    id
                    text
                }
            }
        }
    }
}
```

When I'm using the SDK, I got this exception:

> **`Sdk4me.GraphQL.Sdk4meException`:** Missing required scope(s): agile-board:Read, agile-board-column:Read, team:Read, knowledge-article:Read, problem:Read, product-backlog:Read, project:Read, task:Read, request-template:Read, workflow-template:Read, workflow:Read, release:Read

After searching inside the SDK, I've found this code:

```csharp
namespace Sdk4me.GraphQL
{
    //
    // Summary:
    //     Updates an existing request.
    internal static class RequestUpdateMutation : Mutation<RequestUpdatePayload, RequestUpdateInput>
    {
        //
        // Summary:
        //     Initialize an new RequestUpdate mutation instance.
        internal RequestUpdateMutation(RequestUpdateInput data)
            : base("requestUpdate", "RequestUpdateInput!", data, new HashSet<IQuery> { new RequestQuery
            {
                FieldName = "request",
                IsConnection = false
            }.Select("*") })
        {
        }
    }
}
```

Why is the `Select("*")` method called and is there a way that I only can call `Select(RequestField.Notes)` as the response?

--------------------------------------

As a work around I've tried to create my own extension to pass an array of `RequestField`s

```csharp
namespace MyProject.Extensions;

internal class Sdk4meClientMutationExtension
{
    public static async Task<RequestUpdatePayload> Mutation(this Sdk4meClient client, RequestUpdateInput input, params RequestField[] requestFields)
    {
        return await client.Mutation(new RequestUpdateMutation(input, requestFields)); // <-- error on this line.
    }
}

internal class RequestUpdateMutation : Mutation<RequestUpdatePayload, RequestUpdateInput>
{
    internal RequestUpdateMutation(RequestUpdateInput data, params RequestField[] requestFields)
        : base(
            "requestUpdate",
            "RequestUpdateInput!",
            data, 
            new HashSet<IQuery> {
                new RequestQuery().Select(requestFields) 
            }
        )
    { }
}
```

But then I've seen that the method `Task<TOutEntity> Mutation<TOutEntity, TInEntity>(Mutation<TOutEntity, TInEntity> input)` is defined as internal and that give me a compilation error.

> Argument 2: cannot convert from `MyProject.Extentions.RequestUpdateMutation` to `Sdk4me.GraphQL.RequestUpdateInput`

And now comes my second question: Why isn't this public?

* **.NET version:** 6.0
* **Library version:** 1.0.9

**Comment by klaasvandeweerdt** [comment]
@HeinPauwelyn, I had the same issue earlier this week. The `.Select("*")` causes the issues. Currently, there is no way to control the response fields. Yesterday evening I implemented a change that allows you to specify the response query. This is an example.
```csharp
ConfigurationItemUpdateInput input = new()
{
    ID = "NF1lLnFhL0NpLzUyMDE3MjK",
    SystemID = "a system ID 2"
};

var result = client.Mutation(input, new ConfigurationItemQuery().Select(ConfigurationItemField.SystemID), true).Result;
```
calling
```csharp
public static async Task<ConfigurationItemUpdatePayload> Mutation(this Sdk4meClient client, ConfigurationItemUpdateInput input, ConfigurationItemQuery query, bool throwOnError = true)
{
    return await client.Mutation(new ConfigurationItemUpdateMutation(input, query), throwOnError);
}
```
This allows you to control the response in the same way as writing queries. The `.Select("*")` will also be removed as the data entities are generated on the QA GraphQL schema. This could cause an issue when running it again in a production instance.

This part is still in testing, everything looks fine but it requires additional testing. I do expect to publish a new NuGet with this new functionality this weekend.

On the second question, no specific reason, I probably typed internal once a long time ago.

**Comment by klaasvandeweerdt** [comment]
@HeinPauwelyn,
Thank you again for the feedback and analysis. `.Select("*")` has been replaced with `.Select()` which returns a predefined field selection and you can now also specify the response query for a mutation.
```csharp
var result = client.Mutation(input, new ConfigurationItemQuery().Select(ConfigurationItemField.SystemID)).Result;
```
I published a new release, version 1.1.0, as well as a new NuGet package.
